src/TestSuite/Schema/SchemaCleaner.php

```diff
@@ -0,0 +1,148 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @since         4.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite\Schema;
+
+use Cake\Console\ConsoleIo;
+use Cake\Database\Schema\BaseSchema;
+use Cake\Database\Schema\CollectionInterface;
+use Cake\Datasource\ConnectionInterface;
+use Cake\Datasource\ConnectionManager;
+
+/**
+ * This class will help dropping all tables of a given connection
+ * and truncate the non phinx migrations tables.
+ */
+class SchemaCleaner
+{
+    /**
+     * Drop all tables of the provided connection.
+     *
+     * @param string $connectionName Name of the connection.
+     * @param \Cake\Console\ConsoleIo|null $io Console IO to output the processes.
+     * @return void
+     * @throws \Exception if the dropping failed.
+     */
+    public static function drop(string $connectionName, ?ConsoleIo $io = null)
+    {
+        self::info('Dropping all tables for connection ' . $connectionName, $io);
+
+        $schema = static::getSchema($connectionName);
+        $dialect = static::getDialect($connectionName);
+
+        $stmts = [];
+        foreach ($schema->listTables() as $table) {
+            $table = $schema->describe($table);
+            $stmts = array_merge($stmts, $dialect->dropTableSql($table));
```

Comment:
@markstory I am not sure on how to solve this error. To my understanding, there is an inconsistency in the dropTableSql(), and generally in most of the SchemaDialect's method with a TableSchema and not a TableSchemaInterface as parameter.

How would you solve that?

Reply:
You could use an `instanceof` check to only call dropTableSql on objects that implement the correct interface.

Reply:
Agreed, I'll update accordingly